### DISCUSSION TITLE
Determining a Football leagues lineup structure

### DISCUSSION THREAD

**Discussion by klamott** [cwendt94/espn-api#485] [ESPN API] [answered]
I'm trying to write a function using this library that compares each team's optimal vs actual lineup for a given week. An optimal lineup is the combination of players that, if started, would have scored the highest that week. 

The issue I'm having is how to determine the lineup structure for a league, meaning which player positions are eligible for each lineup slot. For example, how many flex slots are there? Which players are allowed to be started in the flex slots? This is needed for determining the optimal lineup. 

Does this library or ESPN's Football API provide any info for this?

**Comment by dtcarls** [comment, marked as answer]
Here are a few examples. Probably not optimal, but it's how I got it to work. 
https://github.com/dtcarls/fantasy_football_chat_bot/blob/master/gamedaybot/espn/functionality.py#L519
https://github.com/dtcarls/fantasy_football_chat_bot/blob/master/gamedaybot/espn/functionality.py#L592

**Reply by klamott** [reply]
Thank you so much! I realized through your examples I'm missing the basic fact that a flex player's BoxPlayer.slot_position is RB/WR/TE, or whatever the league allows (essentially contains a "/" character and isn't a D/ST).
